lib/protobuf/encoder.ex

```diff
@@ -162,19 +163,18 @@ defmodule Protobuf.Encoder do
   def encode_type(:bool, false), do: encode_varint(0)
   def encode_type({:enum, type}, n) when is_atom(n), do: n |> type.value() |> encode_varint()
   def encode_type({:enum, _}, n), do: encode_varint(n)
-  def encode_type(:float, :infinity), do: <<0, 0, 128, 127>>
-  def encode_type(:float, :negative_infinity), do: <<0, 0, 128, 255>>
-  def encode_type(:float, :nan), do: <<0, 0, 192, 127>>
-  def encode_type(:float, n), do: <<n::32-float-little>>
-  def encode_type(:double, :infinity), do: <<0, 0, 0, 0, 0, 0, 240, 127>>
-  def encode_type(:double, :negative_infinity), do: <<0, 0, 0, 0, 0, 0, 240, 255>>
-  def encode_type(:double, :nan), do: <<1, 0, 0, 0, 0, 0, 248, 127>>
-  def encode_type(:double, n), do: <<n::64-float-little>>
+  def encode_type(:float, :infinity), do: [0, 0, 128, 127]
+  def encode_type(:float, :negative_infinity), do: [0, 0, 128, 255]
+  def encode_type(:float, :nan), do: [0, 0, 192, 127]
+  def encode_type(:float, n), do: [<<n::32-float-little>>]
+  def encode_type(:double, :infinity), do: [0, 0, 0, 0, 0, 0, 240, 127]
+  def encode_type(:double, :negative_infinity), do: [0, 0, 0, 0, 0, 0, 240, 255]
+  def encode_type(:double, :nan), do: [1, 0, 0, 0, 0, 0, 248, 127]
+  def encode_type(:double, n), do: [<<n::64-float-little>>]
 
   def encode_type(:bytes, n) do
-    bin = IO.iodata_to_binary(n)
-    len = bin |> byte_size |> encode_varint
-    <<len::binary, bin::binary>>
+    len = n |> byte_size |> encode_varint
+    len ++ n
```

Comment:
This will ultimately create an improper list, which is not a problem when we call `iodata_to_binary` in the end. Not sure if it is dangerous in other contexts though.

Reply:
The memory usage went up a tad bit. Will leave it like this for now.